Makefile

```diff
@@ -0,0 +1,6 @@
+.PHONY:
+all:
+	@go run internal/mage/main.go -l
+
+.DEFAULT:
+	@go run internal/mage/main.go "$@"
```

Comment:
Dang, I glossed over this but `"$@"` doesn't do what I hoped it would - what it does in Bash - it's just really close. It only takes the first argument, so I can't do this for #3739: `make boot:dev sdk:go:test`:

```
❯ make boot:dev sdk:go:test
not enough arguments for target "Boot:Dev", expected 1, got 0
exit status 2
make: *** [Makefile:6: boot:dev] Error 1
```

In general this limitation makes me wonder if the `Makefile` wrapper is even worth it since we wouldn't be able to use it for targets that take arguments.

Maybe we could just have a `hack/mage` wrapper script?

Reply:
Yeah I feel the same way. `make` had a good run. I think it's time to let it rest.


![](https://y.yarn.co/9073ccdf-70b7-4ca6-9995-b6e6175e8709_text.gif)

Reply:
> Dang, I glossed over this but "$@" doesn't do what I hoped it would - what it does in Bash - it's just really close. It only takes the first argument, so I can't do this for https://github.com/dagger/dagger/issues/3739: make boot:dev sdk:go:test:
> ❯ make boot:dev sdk:go:test

Right -- this means make `boot:dev` AND `sdk:go:test`.

Had the same problem with `sdk:*:publish <argument>` Perhaps a shell script as you suggested would be better.

One weird thing I can't understand is how do we "make" multiple targets with mage? Maybe the answer is we don't.

Reply:
Updated to `./hack/make`